dev-docs/source/MaskWorkspace.rst

```diff
@@ -0,0 +1,101 @@
+=============
+
+.. contents::
+  :local:
+
+**NOTE: This document is may require additional work**
+
+What is it?
+-----------
+
+A :py:obj:`mantid.dataobjects.MaskWorkspace` is a :py:obj:`mantid.dataobjects.SpecialWorkspace2D`
+that may be applied to another Workspace in order to non-destructively exclude data from future calculations.
+It may be reused to apply masking to additional Workspaces as applicable to your workflow.
+
+How Does it Work?
+-----------------
+
+It does so by mapping :term:`Detector ID <Detector ID>`'s to a boolean value.
+
+`0` = Unmasked = LIVE_VALUE
+
+`1` = Masked = DEAD_VALUE
+
+
+This boolean is stored in the first and only y-value of one of many spectra *initially* mapped 1:1 to the Detector ID's of an Instrument. 😐
+
+This is per the implementation of :py:obj:`mantid.dataobjects.SpecialWorkspace2D`.
+
+**NOTE:** This mapping is not *guaranteed* to always be 1:1, as the mapping allows multiple Detector ID to be associated with the same Spectrum.
+However, it is uncommon to group a mask's detector mapping and the MaskWorkspace interface has no methods supporting such an operation.
+(*An example of such a usecase would improve this article. Otherwise MaskWorkspace should actively enforce a 1:1 policy.*)
+
+This mapping seems odd at first, but serves a pragmantic purpose when interfacing with the rest of Workbench.
+*Opinion:* This implementation conflates frontend/backend representation of the data by modeling its format after what a user might expect particle data to be in.
+E.g. Each Detector has a Spectrum with data recorded over some unit, but for a SpecialWorkspace2D this value would be constant.
+
+
+The **X Data/Y Data** of the MaskWorkspace should be the **only source of truth** when determining if a Detector ID is masked.
+
+However, as a :py:obj:`mantid.dataobjects.SpecialWorkspace2D`, it also has a SpectrumInfo->DectectorInfo.
+
+This DetectorInfo also has the capability to track whether or not a given Detector ID is masked.😐
+
+The SpectrumInfo *also* can track if a Detector ID is masked but becomes unreliable for more complex mappings.
+
+**Don't use either of the above.^**
+
+These objects serve as metadata about the Workspace.
+On any other Workspace, these determine if specific Detector IDs or Spectrum are masked.
+In the case of a MaskWorkspace, relying on these instead of the main data creates indirection and uncertainty.
+*Opinion:* The masking aspect of DetectorInfo and SpectrumInfo probably shouldnt have been included on MaskWorkspace to begin with.
+
+
+**NOTE:** Typically, throughout the codebase if a Detector ID *does not exist* in a MaskWorkspace, it is considered *Masked*.
+However, it is safer to fully define and validate that your MaskWorkspace has a value assigned for each Detector ID of your target workspace.
+*Opinion:* There may be an opportunity for performance improvement if instead a mask only held Detector IDs it masked, e.g. the opposite of what is currently assumed.
```

Comment:
The "big" problem is that there are two style of mask
* based on pixel/instrument component - what you are frequently doing
* based on spectrum number - there are a number of workflows where the instrument has been deleted and the user wants to mask things